### ISSUE TITLE
 Logical error: 'Server 2 client 9 already registered in dispatcher'.

### ISSUE THREAD

**Issue by lzydmxy** [JDRaftKeeper/RaftKeeper#107]
### Description
```
2023-11-16T08:08:44.5759742Z 2023.11.16 07:56:43.746655 [ 125 ] <Trace> ForwardingConnectionHandler: Forwarding handler socket readable
2023-11-16T08:08:44.5760136Z 2023.11.16 07:56:43.746676 [ 125 ] <Trace> ForwardingConnectionHandler: forwarding handler socket available
2023-11-16T08:08:44.5760696Z 2023.11.16 07:56:43.746689 [ 125 ] <Trace> ForwardingConnectionHandler: Try handle new package
2023-11-16T08:08:44.5761128Z 2023.11.16 07:56:43.746708 [ 125 ] <Trace> ForwardingConnectionHandler: Receive Handshake
2023-11-16T08:08:44.5761516Z 2023.11.16 07:56:43.746725 [ 125 ] <Trace> ForwardingConnectionHandler: forwarding handler socket available
2023-11-16T08:08:44.5762034Z 2023.11.16 07:56:43.746898 [ 125 ] <Fatal> : Logical error: 'Server 2 client 9 already registered in dispatcher'.
2023-11-16T08:08:44.5762288Z 2023.11.16 07:56:43.747143 [ 11 ] <Trace> BaseDaemon: Received signal 6
```
### How to reproduce 
found in https://github.com/JDRaftKeeper/RaftKeeper/pull/106/checks?check_run_id=18735304047

**Comment by lzydmxy** [comment]
![image](https://github.com/JDRaftKeeper/RaftKeeper/assets/49020515/66ab8204-48df-4a45-b612-f4b3ff3af16d)

**Comment by JackyWoo** [comment]
Fixed by #118
